src/main/java/frc/robot/subsystems/colour_wheel/commands/Spin.java

```diff
@@ -0,0 +1,58 @@
+package frc.robot.subsystems.colour_wheel.commands;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.colour_wheel.ColourWheel;
+
+/**
+ * Spin the wheel 3 times.
+ */
+public class Spin extends CommandBase {
+
+    private final ColourWheel colourWheel;
+    private double power;
+    private String initColour;
+    private int spinCount;
+    boolean flag = true;
+    boolean flag2 = true;
+
+    public Spin(ColourWheel colourWheel, double power) {
+        this.colourWheel = colourWheel;
+        this.power = power;
+        addRequirements(colourWheel);
+    }
+
+    @Override
+    public void initialize() {
+        colourWheel.updateSensor();
+        initColour = colourWheel.getColorString();
+        spinCount = 0;
+        colourWheel.power(power);
+    }
+
+    @Override
+    public void execute() {
+        colourWheel.updateSensor();
+        if (!colourWheel.getColorString().equals(initColour))
+            flag = true;
+        else {
+            if (flag) {
+                flag = false;
+                spinCount++;
+            }
+        }
+        if (spinCount == 5 && flag2) {
+            colourWheel.power(0.5 * power);
+            flag2 = false;
+        }
+    }
+
+    @Override
+    public boolean isFinished() {
+        return spinCount == 6;

```

Comment:
the condition is wrong, after 5 spins, the control panel reset the amount of spin

Reply:
I'm not sure I understand

Reply:
Every 5 spins in the game, the count reset to 0

Reply:
@Se-gol, was this conversation really resolved?